Enable Single Product block

gigitux opened the pull request:
This PR enables the Single Product block. Kudos to @thealexandrelara for working on this block in the last weeks! 🙇 

### Testing

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

⚠️  _For those who review this PR: use a product build (`npm run build:deploy`)_

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

1. Add the Single Product block in a post/page.
2. Visit the created post/page.
3. Click the `add to cart` button .
4. Ensure that the product is added to the cart and any notice is visible.
5. Edit the Single Product Template.
6. Add the Single Product block.
7. Repeat 3-4 steps.

* [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Performance Impact

<!-- Please document any known performance impact (positive or negative) here. If negative, provide some rationale for why this is an okay tradeoff or how this will be addressed. -->

### Changelog

> Enable Single Product Block.

gigitux committed:
Enable Single Product block

gigitux committed:
Merge branch 'trunk' into add/enable-single-product-block

kmanijak reviewed (changes requested):
I think you should also remove the PHP flag [here](https://github.com/woocommerce/woocommerce-blocks/blob/cb11cd2537e35561a9c930f893ee72d82ab6bc75/src/BlockTypesController.php#L227).

Please also remove the entry from the docs section of features behind the feature flag [here](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md?plain=1#L38).

Also, Single Product block is available when `npm run build`, but I think `npm run build:deploy` determines the production build and then the Single Product block is not available (I presume, it's because of the PHP flag).

gigitux committed:
Single Product Block: remove experimental build flag

gigitux committed:
Merge branch 'trunk' of https://github.com/woocommerce/woocommerce-blocks into add/enable-single-product-block

gigitux committed:
Merge branch 'add/enable-single-product-block' of https://github.com/woocommerce/woocommerce-blocks into add/enable-single-product-block

gigitux commented:
> I think you should also remove the PHP flag [here](https://github.com/woocommerce/woocommerce-blocks/blob/cb11cd2537e35561a9c930f893ee72d82ab6bc75/src/BlockTypesController.php#L227).
> 
> Please also remove the entry from the docs section of features behind the feature flag [here](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md?plain=1#L38).
> 
> Also, Single Product block is available when `npm run build`, but I think `npm run build:deploy` determines the production build and then the Single Product block is not available (I presume, it's because of the PHP flag).

You're right! Thanks! I fixed with 91588ff51b6e6ffe3f314acec1fae08a77f4d649. Also, I updated testing instructions.

kmanijak reviewed (approved):
Works as expected! 💪 

Kudos to @thealexandrelara!
